src/test/java/cloud/fogbow/common/plugins/cloudidp/opennebula/OpenNebulaIdentityProviderPluginTest.java

```diff
@@ -0,0 +1,123 @@
+package cloud.fogbow.common.plugins.cloudidp.opennebula;
+
+import cloud.fogbow.common.constants.OpenNebulaConstants;
+import cloud.fogbow.common.exceptions.FogbowException;
+import cloud.fogbow.common.exceptions.InvalidParameterException;
+import cloud.fogbow.common.exceptions.UnauthenticatedUserException;
+import cloud.fogbow.common.exceptions.UnexpectedException;
+import cloud.fogbow.common.models.OpenNebulaUser;
+import cloud.fogbow.common.util.connectivity.cloud.opennebula.OpenNebulaClientUtil;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.BDDMockito;
+import org.mockito.Mockito;
+import org.opennebula.client.Client;
+import org.opennebula.client.ClientConfigurationException;
+import org.opennebula.client.OneResponse;
+import org.opennebula.client.user.UserPool;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.HashMap;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({OpenNebulaClientUtil.class})
+public class OpenNebulaIdentityProviderPluginTest {
+
+    private OpenNebulaIdentityProviderPlugin plugin;
+    public static final String FAKE_IDENTITY_URL = "http://localhost:0000";
+    public static final String FAKE_USER_ID = "fake-id";
+    public static final String FAKE_USER_PASSWORD = "fake-password";
+    public static final String INVALID_PASSWORD = "invalid-password";
+    public static final String ONE_RESPONSE_MESSAGE = "Random oneResponse return message";
+
+    @Before
+    public void setup() throws UnexpectedException {
+        this.plugin = Mockito.spy(new OpenNebulaIdentityProviderPlugin(FAKE_IDENTITY_URL));
+
+        Client client = Mockito.mock(Client.class);
+
+        PowerMockito.mockStatic(OpenNebulaClientUtil.class);
+        BDDMockito.given(OpenNebulaClientUtil.createClient(Mockito.anyString(), Mockito.anyString())).willReturn(client);
+    }
+
+    @Test
+    public void testAuthenticateSuccess() throws FogbowException {
+        // setup
+        Mockito.doReturn(true).when(this.plugin).isAuthenticated(Mockito.anyString());
+        HashMap<String,String> credentials = getCredentials(FAKE_USER_ID, FAKE_USER_PASSWORD);
```

Comment:
why are you not using the Map interface here?

Reply:
Done.